### PR TITLE
Establish data structure

### PR DESCRIPTION
**Pull request by glatimer**
This is the basic structure anticipated from the beginning.  eslint enforces using interfaces for object structures so that was the largest change from the original plan. As we learn more about how certain components work, objects may need to be added, removed, or altered (i.e. navMap).

### CODE REVIEW

File: src/dataStructures.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// hotSpotData: types of media resources for a hotspot within a photosphere
+export type hotspotData = img | audio | video | URL | doc | photosphereLink;
+
+// media objects
+export interface img {
+  tag: "img";
+  src: string;
+  hotspot: img2d | img3d; // can be 2d|3d
```
Comment (mnxn) [inline src/dataStructures.ts:36]: A 2D image can only have 2D hotspots on it, so this field should be `hotspots: Hotspot2D[]`.

File: src/dataStructures.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+export interface navMap {
+  src: string;
+  hotspot: hotspotData[];
+}
+
+// Photosphere: a single 360-environment
+export interface photosphere {
+  id: string;
+  src: string;
+  hotspot: hotspotData[];
```
Comment (mnxn) [inline src/dataStructures.ts:25]: The hotspot data type doesn't contain any information about the hotspot's position.
It should be an array of 3D hotspots instead.

**Review by mnxn — CHANGES_REQUESTED** [review]
The popular convention is to use PascalCase for type/interface naming in TypeScript. For example: `Photosphere` and `PhotosphereLink`.

I also left some comments on specific things that can be improved.

File: src/dataStructures.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    Virtual Field Environment Software
+   -----------------------------------------------------------------------
+    Instantiation of the objects that will be used for interacting with 
+    Photo Sphere Viewer to build a virtual geology field guide.
+   ----------------------------------------------------------------------- */
+
+// Virtual Field Environment: the total collection of photo sphere environments
+export interface VFE {
+  name: string;
+  map: Map<string, number>; // arguments <"imgURL", center_number>
```
[the remark this answers is not in the document]
Comment (glatimer) [reply src/dataStructures.ts:11]: So you're saying to switch the names for VFE and navMap?

[the remark this answers is not in the document]
Comment (mnxn) [reply src/dataStructures.ts:11]: No, the map _field_ should be `map : NavMap`.

[the remark this answers is not in the document]
Comment (glatimer) [reply src/dataStructures.ts]: aha. That makes sense.

File: src/dataStructures.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  hotspot: HotSpot3d[];
+  backgroundAudio: string;
+}
+
+// hotspot: a clickable resource that is 2d (x, y)
+export interface HotSpot2d {
+  x: number;
+  y: number;
+  toolTip: string;
+  hotspot: HotSpot2d;
```
Comment (mnxn) [inline src/dataStructures.ts:36]: Shouldn't this line be something like `data: HotSpotData` instead?
Sames goes for the other hotspot type.

Comment (glatimer) [reply src/dataStructures.ts:36]: I'm not quite sure how we will be using these objects it seems lol.  Fixing now!

Comment (glatimer) [reply src/dataStructures.ts:36]: Oh, I get it.  Similarly, to how we added hotspot markers.

**Review by mnxn — APPROVED** [review]
Thank you for making all the changes!

Looks good to me now.

**Pull request merged**